Adding Debug Info for RRef Context

mrshenli opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #30645 Apply clang-format to rpc/utils.cpp
* **#30610 Adding Debug Info for RRef Context**
* #30609 Add RRef.__str__() API



Differential Revision: [D18763592](https://our.internmc.facebook.com/intern/diff/D18763592)

mrshenli committed:
[Don't Merge] For debugging only

[ghstack-poisoned]

mrshenli committed:
Update on "[Don't Merge] For debugging only"


Differential Revision: [D18763592](https://our.internmc.facebook.com/intern/diff/D18763592)

[ghstack-poisoned]

mrshenli committed:
Update on "[Don't Merge] For debugging only"


Differential Revision: [D18763592](https://our.internmc.facebook.com/intern/diff/D18763592)

[ghstack-poisoned]

mrshenli committed:
Update on "[Don't Merge] For debugging only"


Differential Revision: [D18763592](https://our.internmc.facebook.com/intern/diff/D18763592)

[ghstack-poisoned]

mrshenli committed:
Update on "Adding Debug Info for RRef Context"




Differential Revision: [D18763592](https://our.internmc.facebook.com/intern/diff/D18763592)

[ghstack-poisoned]

mrshenli committed:
Update on "Adding Debug Info for RRef Context"




Differential Revision: [D18763592](https://our.internmc.facebook.com/intern/diff/D18763592)

[ghstack-poisoned]

mrshenli committed:
Update on "Adding Debug Info for RRef Context"




Differential Revision: [D18763592](https://our.internmc.facebook.com/intern/diff/D18763592)

[ghstack-poisoned]

mrshenli committed:
Update on "Adding Debug Info for RRef Context"




Differential Revision: [D18763592](https://our.internmc.facebook.com/intern/diff/D18763592)

[ghstack-poisoned]

bddppq commented on test/rpc_test.py:
```diff
@@ -1179,6 +1188,59 @@ def test_rref_str(self):
+                rank=self.rank,
+                world_size=self.world_size,
+            )
+
```
add a `dist.barrier()` here to make sure all pg initialization are completed?

mrshenli replied:
This should be fine as `init_process_group` itself is a barrier.

mrshenli committed:
Update on "Adding Debug Info for RRef Context"




Differential Revision: [D18763592](https://our.internmc.facebook.com/intern/diff/D18763592)

[ghstack-poisoned]

mrshenli committed:
Update on "Adding Debug Info for RRef Context"




Differential Revision: [D18763592](https://our.internmc.facebook.com/intern/diff/D18763592)

[ghstack-poisoned]
